src/ui_basic/panel.cc

```diff
@@ -284,12 +284,14 @@ void Panel::do_redraw_now(const std::string& message) {
 		g_mouse_cursor->change_cursor(app->is_mouse_pressed());
 		g_mouse_cursor->draw(rt, app->get_mouse_position());
 
-		if (modal_ != nullptr) {
-			modal_->do_tooltip();
-		} else if (is_modal()) {
-			do_tooltip();
-		} else {
-			ff.do_tooltip();
+		if (message.empty() && (flags_ & pf_hide_all_overlays) == 0) {
```

Comment:
Can we have a comment why and where the tooltips are finally shown?
Especially message.empty() confuses me.